TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java

```diff
@@ -41,12 +46,24 @@ private static Settings getSettings(@NonNull String configFile) {
     }
 
     private static IConnectionCaptureFactory getConnectionCaptureFactory(String traceLogsDirectory) {
-        return new FileConnectionCaptureFactory(traceLogsDirectory);
+        if (traceLogsDirectory == null) {
+            System.err.println("No trace log directory specified.  Logging to /dev/null");
+            return new IConnectionCaptureFactory() {
+                @Override
+                public IChannelConnectionCaptureSerializer createOffloader(String connectionId) throws IOException {
+                    return new StreamChannelConnectionCaptureSerializer(connectionId, () ->
+                            CodedOutputStream.newInstance(NullOutputStream.getInstance()),
+                            cos -> CompletableFuture.completedFuture(null));
+                }
+            };
+        } else {
+            return new FileConnectionCaptureFactory(traceLogsDirectory);
+        }
     }
 
     public static void main(String[] args) throws InterruptedException, IOException {
 
-        String traceLogsDirectory = args[0];
+        String traceLogsDirectory = args.length > 0 ? args[0] : null;
```

Comment:
Then there are no logs - see lines 51-58.  An error message is also emitted.
Ideally, the log sinks will eventually be handled via SPI.  Once our Kafka publisher comes into play, we'll soon want to reconsider sink configuration, but I'm pushing that off a bit to focus on other parts of the system.

The null-logger was put into place so that we can benchmark the minimal overhead that this proxy will create.